tests/Support/Constraint/CssConstraint.php

```diff
@@ -134,26 +164,22 @@ protected static function getCssRegularExpressionMatcher(string $css): string
      */
     private static function getCssRegularExpressionReplacement(array $matches): string
     {
-        if (($matches[1] ?? '') !== '') {
-            $regularExpressionEquivalent = '(?:\\s*+;)?+\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
-        } elseif (($matches[2] ?? '') !== '') {
-            $regularExpressionEquivalent = '\\s*+' . \preg_quote($matches[2], '/') . '\\s*+';
-        } elseif (($matches[3] ?? '') !== '') {
-            $regularExpressionEquivalent = '\\s*+';
-        } elseif (($matches[4] ?? '') !== '') {
-            $regularExpressionEquivalent = \preg_quote($matches[4], '/') . '\\s*+';
-        } elseif (($matches[5] ?? '') !== '') {
-            $regularExpressionEquivalent = '\\s++';
-        } elseif (($matches[6] ?? '') !== '') {
-            $regularExpressionEquivalent = '(?i:' . \preg_quote($matches[6], '/') . ')';
-        } elseif (($matches[8] ?? '') !== '' || ($matches[10] ?? '') !== '') {
-            $urlMatcher = '(?:([\'"]?+)' . \preg_quote(($matches[8] ?? '') . ($matches[10] ?? ''), '/') . '\\g{-1})';
-            $regularExpressionEquivalent = '@import\\s++(?:' . $urlMatcher . '|url\\(\\s*+' . $urlMatcher . '\\s*+\\))';
-        } elseif (($matches[12] ?? '') !== '') {
-            $regularExpressionEquivalent = 'url\\(\\s*+([\'"]?+)' . \preg_quote($matches[12], '/') . '\\g{-1}\\s*+\\)';
-        } elseif (($matches[13] ?? '') !== '') {
-            $regularExpressionEquivalent = '0?+\\.';
-        } else {
+        $regularExpressionEquivalent = null;
+
+        foreach (self::CSS_REGULAR_EXPRESSION_REPLACEMENTS as $index => $replacement) {
+            if (($matches[$index] ?? '') !== '') {
+                $regularExpressionEquivalent = \preg_replace_callback(
+                    '/\\$(\\d++)/',
+                    static function (array $referenceMatches) use ($matches): string {
+                        return \preg_quote($matches[(int)$referenceMatches[1]] ?? '', '/');
+                    },
+                    $replacement
+                );
+                break;
+            }
+        }
+
+        if ($regularExpressionEquivalent === null) {
```

Comment:
I'd prefer `!is_string` here to make this check more specific (for the benefit of the reader and static code analysis).

Reply:
OK.  Changed.